src/components/FilterControls/SortingGroup.vue

```diff
@@ -0,0 +1,64 @@
+<script>
+  import { pipe, reverse } from "ramda";
+
+  import mediator from "@/mediator";
+  import SortOption from "@/components/FilterControls/SortOption";
+  import { sortRecentlyPrepared, sortById } from "@/business/meals";
+
+  export default {
+    components: {SortOption},
+    name: 'SortingGroup',
+    props: {
+      meals: {
+        type: Array,
+        default: () => []
+      }
+    },
+    data: function() {
+    	return {
+    		selected: ""
+      }
+    },
+    methods: {
+    	sortRecentlyPrepared,
+      sortLongestUnused: function(meals) {
+    		return pipe(
+    			sortRecentlyPrepared,
+          reverse
+        )(this.meals)
+      },
+      handleSorting: function(sortName, sorter) {
+    		this.selected = sortName;
+    		mediator.$emit("sorted", sorter(this.meals));
+      },
+      clearSorting: function() {
+    		this.selected = "";
+        mediator.$emit("sorted", sortById(this.meals));
+      }
+    },
+    render: function(h) {
+      return (
+      	<div class="col-xs-12 col-md-9">
+          <sort-option
+            active={ this.selected === 'recentlyPrepared' }
+            sortName="recentlyPrepared"
+            sorter={ this.sortRecentlyPrepared }
```

Comment:
passing this as a prop is a bit kludgy and unnecessary. there's a few other ways I could have done this. I might refactor.

Reply:
One way of simplifying this might be to generate a little hash that contains the sorters + the sorting method:
```javascript
const SORTERS = {
  recentlyPrepared: {
    text: 'recently used',
    sort: sortRecentlyPrepared
  },
  longestUnused: {
    text: 'longest unused',
    sort: pipe(sortRecentlyPrepared, reverse)
  }
}
```

Then when you render, you'd do something like:
```javascript
{SORTERS.keys().map(key =>
  <sort-option
    active={this.selected === key}
    sortName={key}
    onSelected={SORTERS[key].sort}
    onUnselected={this.clearSorting}
  >{SORTERS[key].text}</sort-option>
}
```
The benefit is that now all of the logic is encapsulated here - and you don't have to pass down a `sorter`. The child component can simply `$emit` `un|selected` - and the parent has already pre-bound what to do.

Reply:
I would definitely do this if my filters list were dynamic. But since there are only two static filters, saving this sort of configuration seemed like a lot of over-engineering and misdirection, But agreed, it's definitely a useful pattern.

Re: not passing down the sorter - I didn't have to do that here either. There are a few ways I could have bound the sorting functions the events in the SortOption component. I just picked this one because....i don't know. I think i was being willfully self damaging. I literally didn't love this idea even as i was considering writing it, and yet i used it anyways. Not sure what that says about me.

Reply:
Yeah, I get that. I like building out the above type of abstraction even for short lists because I like that it also gives me a place to cache bound functions and such. I made some comments on your React version that probably relate more to it than this spot, though. :)

I wouldn't have even mentioned, here, except that you specifically asked for suggestions on other patterns :)

Reply:
yep, and it's a great comment :) I think the takeaway is that there are two valid patterns: one tuned to small sets of static data; one appropriate for dynamic content or more complex mappings. Choose wisely, according to your needs :)